lib/statsample-glm/glm/base.rb

```diff
@@ -74,14 +105,55 @@ def fitted_mean_values
         @regression.fitted_mean_values
       end
 
+      # Returns the residual for every data point
+      #
+      # @return [Vector] all residuals in a vector
+      #
+      # @example
+      #   require 'statsample-glm'
+      #   data_set = Daru::DataFrame.from_csv "spec/data/logistic.csv"
+      #   glm  = Statsample::GLM.compute data_set, "y", :logistic, {constant: 1}
+      #   glm.residuals
+      #     # #<Daru::Vector:22263420 @name = y @metadata = {} @size = 50 >
+      #     #                                         y
+      #     #                   0 -0.18632025624516532
+      #     #                   1  -0.5146459448198846
+      #     #                   2     0.15916476717451
+      #     #                   3  -0.9241524337773334
+      #     #                   4   0.2281471136368174
+      #     #                 ...                  ...
+      #
       def residuals
         @regression.residuals
       end
 
+      # Returns the degrees of freedom value.
+      #
+      # @return [Integer] the degrees of freedom
+      #
+      # @example
+      #   require 'statsample-glm'
+      #   data_set = Daru::DataFrame.from_csv "spec/data/logistic.csv"
+      #   glm  = Statsample::GLM.compute data_set, "y", :logistic, constant: 1
+      #   glm.degree_of_freedom
+      #     # => 47
+      #
       def degree_of_freedom
         @regression.degree_of_freedom
```

Comment:
I agree. I think, the plural "degrees" is more correct.

Similarly, maybe we should change `#standard_error` to the plural form `#standard_errors`, because it returns a vector of standard errors rather than just one value. What do you guys think?